Update active storage guide to match recent PRs [ci-skip] (#42735)
rails/rails: guides/source/active_storage_overview.md

backups and migrations.

> Using Active Storage, an application can transform image uploads with

Reviewer: If you go with the "parenthesis" suggestion:

```suggestion
Using Active Storage, an application can transform image uploads 
```

> [ImageMagick](https://www.imagemagick.org) or [libvips](https://github.com/libvips/libvips), 
> generate image representations of non-image uploads like PDFs and videos using 
> [poppler](https://poppler.freedesktop.org/) and [ffmpeg](http://ffmpeg.org/), and extract 
> metadata from arbitrary files.

Reviewer: This sentence is getting a bit cumbersome...  Perhaps parenthesis can help?

```suggestion
(via [ImageMagick](https://www.imagemagick.org) or [libvips](https://github.com/libvips/libvips)), 
generate image representations of non-image uploads like PDFs (via [poppler](https://poppler.freedesktop.org/)) 
and videos (via [ffmpeg](http://ffmpeg.org/)), and extract metadata from arbitrary files.
```

Or perhaps we should just forego mentioning all these tools here?

```suggestion
generate image representations of non-image uploads like PDFs and videos, and extract 
metadata from arbitrary files.
```

Author: I'd considered removing all of them. It felt like too much detail about how things are implemented for an opening section, and the "Analyzing Files" section already describes the tools required. 

However, the "Transforming Images" only mentions the image processing gem. Maybe we should add a "requirements" section near the top and mention the necessary gems, libs, etc?

Reviewer: There is a note about FFmpeg and Poppler under the [Previewing Files](https://guides.rubyonrails.org/v6.1/active_storage_overview.html#previewing-files) section.  Perhaps, in combination with the other sections, that is sufficient?  Although I'm not opposed to a "Requirements" section either.

Author: I'm a bit bothered by the inconsistency. 

Previewing: Big honking red alert about third-party applications needed;
Transforming: A brief mention about adding the `image_processing` gem to the Gemfile but not of the applications;
Analysing: A full mention of all gems and tools, but no mention about adding them to the Gemfile;

I'd rather remove the gems and tools from all three sections above and only mention them in "Requirements" probably as subsection 1.1. I'd also not mention `mini_magick` or `ruby-vips` gems and focus on `image_processing` which already will add both.

```
1.1 Requirements

Various features of Active Storage depend on third party applications which Rails will not install, and you must include separately:

* `ImageMagick` or `libvips` for image analysis and transformations;
* `poppler` or `muPDF` for PDF previews;
* `ffmpeg` for video/audio analysis and video previews;

To enable image transformations the analysis the `image_processing` gem is also necessary:

'''ruby
gem 'image_processing'
'''
```

Reviewer: Yes, something like that would be good, though we should be sure to include the warning about licensing implications.  If you'd like we can handle all of that in a follow-up PR, to keep this PR simple.

Active Storage analyzes files once they've been uploaded by queuing a job in Active Job. Analyzed files will store additional information in the metadata hash, including `analyzed: true`. You can check whether a blob has been analyzed by calling [`analyzed?`][] on it.

> Image analysis provides `width` and `height` attributes. Video analysis provides these, as well as `duration`, `angle`, `display_aspect_ratio` and presence/absence of `video` and `audio` channels. Audio analysis provides `duration` and `bit_rate` attributes.

Reviewer: ```suggestion
Image analysis provides `width` and `height` attributes. Video analysis provides these, as well as `duration`, `angle`, `display_aspect_ratio`, and `video` and `audio` booleans to indicate the presence of those channels. Audio analysis provides `duration` and `bit_rate` attributes.
```

Author: Ooooops. Sorry about missing that. It's fixed now.

Active Storage analyzes files once they've been uploaded by queuing a job in Active Job. Analyzed files will store additional information in the metadata hash, including `analyzed: true`. You can check whether a blob has been analyzed by calling [`analyzed?`][] on it.

> Image analysis provides `width` and `height` attributes. 
> Video analysis provides these, as well as `duration`, `angle`, `display_aspect_ratio` and presence/absence of `video` and `audio` channels.

Reviewer: ```suggestion
Image analysis provides `width` and `height` attributes. Video analysis provides these, as well as `duration`, `angle`, `display_aspect_ratio`, and `video` and `audio` booleans to indicate the presence of those channels.
```

Author: I had added a line break here because the "Audio analysis" was in a new line. Should I remove the new line from that one too?

Reviewer: Yes, looks good! :+1:

Reviewer: Oops, I was only inspecting the line break...  In the above suggestion, though, I had added an oxford comma and slightly rephrased the last item under video analysis.  If you think that sounds good, could you include those changes too?

Audio analysis provides `duration` and `bit_rate` attributes.

> Image analysis requires the `mini_magick` gem and the ImageMagick library, or the `ruby-vips` gem and the libvips library. 
> Audio/Video analysis requires the [FFmpeg](https://www.ffmpeg.org/) library.
> 
> All three libraries mentioned above must be installed separately, as they are not included in the gems or rails.

Reviewer: Since FFmpeg isn't really a library, what about this arrangement:

```suggestion
Image analysis requires either the `mini_magick` gem and the ImageMagick library, or the `ruby-vips` gem and the libvips library. Note that the libraries are not included in the gems, and must be installed separately.

Audio/Video analysis requires [FFmpeg](https://www.ffmpeg.org/), which must also be installed separately.
```

Author: I'm always confused about what's a library and what's not 😅

Reviewer: Well I guess ImageMagick isn't really a library either, though I often hear it referred to as such.  (A library being something like a `.so` file on Linux or a `.dll` file on Windows.)  I think what we have is fine though. :+1:
